packages/block-library/src/social-links/edit.js

```diff
@@ -58,23 +59,33 @@ export function SocialLinksEdit( props ) {
 
 	const {
 		iconBackgroundColorValue,
+		customIconBackgroundColor,
 		iconColorValue,
 		openInNewTab,
 		size,
 		layout,
 	} = attributes;
+
 	const usedLayout = layout || getDefaultBlockLayout( name );
 
 	// Remove icon background color if logos only style selected.
-	const logosOnly =
-		attributes.className?.indexOf( 'is-style-logos-only' ) >= 0;
+	const logosOnly = attributes.className
+		? attributes.className.indexOf( 'is-style-logos-only' ) >= 0
+		: false;
 	useEffect( () => {
 		if ( logosOnly ) {
+			setbackgroundBackup( {
+				iconBackgroundColor,
+				iconBackgroundColorValue,
+				customIconBackgroundColor,
+			} );
 			setAttributes( {
 				iconBackgroundColor: undefined,
 				customIconBackgroundColor: undefined,
 				iconBackgroundColorValue: undefined,
 			} );
+		} else {
+			setAttributes( { ...backgroundBackup } );
 		}
 	}, [ logosOnly, setAttributes ] );
```

Comment:
As is, the effect could be considered to be missing dependencies (`iconBackgroundColor` and the other two that are put into backup). But I've tested and I can't make it misbehave (i.e. backup stale values). If we add the dependencies it would require more conditional logic to avoid breaking the effect. There are other options that avoid referencing those variables inside the hook but I'm not sure any of that complexity is worth it when this seems okay for now.

Reply:
@stokesman thanks for the feedback but I think its okay of now.

Reply:
Yes, I was saying as much. Still, I took another look and found a pretty simple way to have it work with the added dependencies. I've left that as separate comment/suggestion.

Reply:
Thank you soo much @stokesman for the feedbacks as I am new at contribution and I learned some new things by your comments.